File: lib/ApiGithub.js

class ApiGithub {
    constructor(githubID, repoName) {
        this.githubID = githubID;
        this.repoName = repoName;
    }

    async get(url) {
        let token = process.env.NEXT_PUBLIC_GITHUB_TOKEN;
        let r = await fetch(url, {headers: {
            'Authorization': `token ${token}`
        }});
        let j = await r.json();
        return j;
    }

    async getRepositoryInfo() {
        let j = await this.get(`https://api.github.com/orgs/${this.githubID}/repos`);
        let repoInfo = j.filter(x => x.name === this.repoName)[0];
        return repoInfo;
    }

    async getPullRequests(state) {
        let j = await this.get(`https://api.github.com/repos/${this.githubID}/${this.repoName}/pulls`)

        if(state) {
            j = j.filter(x => x.state === state);
        }

        return j;
    }

    async lsTree(path, branch='main') {
        let j = await this.get(`https://api.github.com/repos/${this.githubID}/${this.repoName}/git/trees/${branch}`);

        let tree = j.tree;

        if(path) {
            tree = tree.filter(x => x.path === path);
        }

        return tree;
    } 

    getRawContentURL(branch, file_path) {
        return `https://raw.githubusercontent.com/${this.githubID}/${this.repoName}/${branch}/${file_path}`;
    }
}

export default ApiGithub;

File: scripts/api_github.py
import requests

class ApiGithub:
    def __init__(self, githubID, repoName):
        self.githubID = githubID;
        self.repoName = repoName;

    def get(self, url):
        NEXT_PUBLIC_GITHUB_TOKEN='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'
        token = NEXT_PUBLIC_GITHUB_TOKEN;

        r = requests.get(url, headers = { 'Authorization': f'token {token}' });
        j = r.json()
        return j;

    def getRepositoryInfo(self):
        j = self.get(f'https://api.github.com/orgs/{self.githubID}/repos');
        repoInfo = list(filter(lambda x: x['name'] == self.repoName, j))[0]
        return repoInfo

    def getPullRequests(self, state):
        j = self.get(f'https://api.github.com/repos/{self.githubID}/{self.repoName}/pulls')

        if state:
            j = list(filter(lambda x: x['state'] == state, j))

        return j

    def lsTree(self, path, branch='main'):
        j = self.get('https://api.github.com/repos/{self.githubID}/{self.repoName}/git/trees/{branch}?recursive=1')

        tree = j.tree

        if path:
            tree = list(filter(lambda x: x['path'] == path, tree))

        return tree

    def getRawContentURL(self, branch, file_path):
        return f'https://raw.githubusercontent.com/{self.githubID}/{self.repoName}/{branch}/{file_path}'